Adding exercises for Wintersession 2024 vim workshop

biermanr opened the pull request:
I added a few exercises here and there to work through with the vim class

biermanr committed:
adding exercises into the READMEs

biermanr committed:
trying blockquotes for exercises

biermanr committed:
more block quotes for exercises

biermanr committed:
needed to remove "missing" letter

biermanr committed:
from stdin and vi *.txt added

biermanr committed:
simplifying ~/.vimrc setup

biermanr committed:
small quotes/pytest/refact adjusts

troycomi commented on 0_intro/README.md:
```diff
@@ -230,8 +245,53 @@ to fill in parts of the expanded text, e.g. function name, arguments, etc.
+> that would make the lines easier to read
+> and it shouldn't be too hard to do by yanking the empty line
+> pasting it where required and then moving down to the next line
+> and finally using the dot commadand `.` to repeat it as much
```
Typo here that seems unintentional `commadand`

troycomi commented on 1_files_windows/README.md:
```diff
@@ -54,22 +54,70 @@ C-ws                horizontal split
+You can open multiple files at once with vim from the command line using
+wildcards such as `vi *.txt`. By default, this does not open each file in a
+separate tab, and instead you can move from one file to the next with :n or :N.
+You can also use `:tab all` to open each buffer in a separate tab.
```
Would probably be a good place to mention the `-o`, `-O`, and `-p` options here for dealing with multiple files.

biermanr replied:
I didn't know about these command-line flags, really cool for opening a handful of files! I added a few sentences description to introduce these

troycomi commented on 1_files_windows/subdir/sub.txt:
```diff
@@ -1 +1 @@
-sub file
+This is sub.txt in the subdir
```
I like these series of changes!  A lot nicer for the exercises.

troycomi commented on 5_change_quotes/README.md:
```diff
@@ -53,12 +54,31 @@ let @b="yi[ca[{0}"
 [vim-surround](https://github.com/tpope/vim-surround) plugin solves this nicely
 and plays well with the `.` command. (use `gx` to go to a url in a file)
+
+## Registers in general
```
These should be formatted to fit into 80 characters

troycomi commented on 6_pytest_output/README.md:
````diff
@@ -75,9 +82,16 @@ performing the replacement.
 Overall we have
 ```
-vi(             select in parenthesis
+vip             select the paragraph
````
Curious about this change.  I think `vip` will get everything in a block of text, e.g. between new lines.  `vi(` should be more specific

biermanr replied:
I thought we wanted to read in the `:r output.tsv` and change the tabs to commas. After the `:r` command we have loaded in that block of text which we can select with `vip` before doing the substitute command. I don't see where the parenthesis are, maybe I'm not looking at the right text?

troycomi replied:
I may also be thinking of something else... You're right here!

troycomi commented on README.md:
````diff
@@ -50,15 +52,39 @@ You need to have vim installed.  While there shouldn't be any version-specific
-```
-We will cover a basic config in the next section
+
+# Setting up your new .vimrc
````
Much nicer way to get their vimrc synced!

troycomi commented on class_vimrc.txt:
```diff
@@ -0,0 +1,36 @@
+set hidden                  "allow modified buffers to hide
+
+set number                  "Display line numbers incrementing
+"set number relativenumber  "Display line numbers in relative positions
```
You should give relativenumber a try.  It makes it much easier to go like 4 or 38 lines up/down since the column shows how many lines away you are.  Much faster than say 755g.

It also makes it much easier to move/copy text.  If I want lines 3-5 above me, you `:-3,-5t.` to make a copy (m to move).  Again, faster than typing the line numbers or counting rows.  Personal taste though!

biermanr replied:
I need to learn this! ok I've made the default to be relative line numbering in the class vimrc

biermanr committed:
edits from PR comments and long-line fixes

troycomi commented on class_vimrc.txt:
```diff
@@ -19,8 +19,8 @@ set incsearch               "highlight while typing
-set number                  "Display line numbers incrementing
-"set number relativenumber  "Display line numbers in relative positions
+set number relativenumber   "Display line numbers in relative positions
+"set number                 "Display line numbers incrementing
```
You actually need both set.  With just relative number, the cursor position is labeled with `0`.  With both, the current line has the line number while the rest are relative.

troycomi committed:
Update class_vimrc.txt

Need both set.  With just relative the current line is labeled 0.  With both the current line has the line number and others are relative.
